packages/office-ui-fabric-react/src/components/HoverCard/examples/HoverCard.Target.Example.tsx

```diff
@@ -53,10 +54,15 @@ const onRenderExpandedCard = (item: IExampleItem): JSX.Element => {
     </div>
   );
 };
-const onRenderItemColumn = (item: IExampleItem, index: number, column: IColumn): JSX.Element | React.ReactText => {
+
+/**
+ * Used for the hoverable "key" cell. In this case, the implementation uses hooks to control
+ * open/closed state, so it must be a function component (not just code within a render callback).
+ */
+const KeyCellWithHoverCard: React.FunctionComponent<{ item: IExampleItem }> = ({ item }) => {
```

Comment:
In this case the rule caught an actual issue: `onRenderItemColumn` (which is not a component or a hook) contained hook calls. I think it happened to work okay in this case, but in general that's not recommended since it can lead to hooks being called conditionally or other strange, hard-to-find issues. 

As a fix, I extracted the logic that needs hooks into this sub-component. (@czearing FYI)